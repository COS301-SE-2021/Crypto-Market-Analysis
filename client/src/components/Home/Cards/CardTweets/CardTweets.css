body{
    margin-top:20px;
    background:#ddd;
}

.widget .panel, .widget.panel {
    border-bottom-width: 1px;
    border-bottom-color: #cfd9db;
    overflow: hidden;
}

.widget {
    margin-bottom: 20px;
}

.panel {
    position: relative;
    border-width: 1px;
    border-color: #cfd9db;
    border-radius: 3px;
    -webkit-box-shadow: 0 .0625rem .0625rem rgba(0,0,0,.05);
    box-shadow: 0 .0625rem .0625rem rgba(0,0,0,.05);
}

.panel-body {
    position: relative;
}

.mb0 {
    margin-bottom: 0!important;
}

.text-info {
    color: #22A7F0!important;
}

.list-table {
    display: table;
    table-layout: fixed;
    width: 100%;
    margin: 0;
    padding: 0;
}

.list-table>li:first-child {
    padding-left: 0;
}

.list-table>li {
    display: table-cell;
    table-layout: fixed;
    vertical-align: middle;
    width: auto;
    padding: 0 5px;
}

.img-bordered-primary {
    -webkit-box-shadow: 0 0 0 2px #fff,0 0 0 4px #446CB3;
    box-shadow: 0 0 0 2px #fff,0 0 0 4px #446CB3;
}

.img-circle {
    border-radius: 50%!important;
}

.ellipsis {
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
}
.semibold {
    font-weight: 600;
}
.nm {
    margin: 0!important;
}