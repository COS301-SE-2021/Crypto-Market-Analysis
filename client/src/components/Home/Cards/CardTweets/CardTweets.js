import React from "react";
import PropTypes from "prop-types";
import "./CardTweets.css"

export default function CardTweets({
    tweetOwner,
    tweetContent,
   
  }) {
    return (
      < >
        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css"/>
       
        <div data-testid="tweet-card" className=" flex flex-col min-w-0 break-words bg-white rounded mb-6 xl:mb-0 shadow-lg">
          <div className="flex-auto p-3">
            <div className="flex flex-wrap">
              <div className="relative w-full pr-4 max-w-full flex-grow flex-1">
                <h5 className="text-blueGray-700 uppercase font-bold text-l">
                 {tweetOwner}
                </h5>
                
              </div>
              <div className="relative w-auto pl-4 flex-initial">
                  <h3 className="mb0">
                    <i className="fa fa-twitter text-info pull-right"></i>
                  </h3>
              </div>
            </div>
                  
            <div className="relative w-full pr-4 max-w-full flex-grow flex-1">
                <h5 className="text-blueGray-700  font-bold text-m">
                  Tweet
                </h5>
              </div>
            <ul className="list-group">
                <li className="list-group-item">
              <p className="nm">{tweetContent}</p>
                <small className="text-muted">1 day ago</small>
                </li>
                {/* <li class="list-group-item">
                <p class="nm">Excepteur sint occaecat cupidatat non proident, sunt in culpa qui <a href="#">@officia</a> deserunt mollit anim id est laborum.</p>
                <small class="text-muted">2 day ago</small>
                </li>
                <li class="list-group-item">
                <p class="nm">Duis aute irure dolor in <a href="#">tweet.er</a> in voluptate velit esse cillum dolore eu fugiat nulla pariatur.</p>
                <small class="text-muted">3 day ago</small>
                </li> */}
            </ul>
          </div>
        </div>
      </>
    )
  }

  CardTweets.defaultProps = {
    tweetOwner:"Tweet Owner",
    tweetContent:"My first Tweet"
  };
  
  CardTweets.propTypes = {
    tweetOwner: PropTypes.string,
    tweetContent: PropTypes.string
  };