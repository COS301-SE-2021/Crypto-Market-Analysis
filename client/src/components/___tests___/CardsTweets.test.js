import {cleanup, render, screen} from '@testing-library/react';
import CardTweets from '../Home/Cards/CardTweets/CardTweets'
import renderer from "react-test-renderer";
import React from "react";

afterEach(()=>{
    cleanup();
})

test('should render tweet component',()=>{
    const tweetInfo = { tweetOwner:"Elon Musk",tweetContent:"Bitcoin is cool"}
    render(<CardTweets tweetOwner={tweetInfo.tweetOwner} tweetContent={tweetInfo.tweetContent}/>)
    const tweet  = screen.getByTestId('tweet-card');
    expect(tweet).toBeInTheDocument();
})

test("matches snapshot",()=>{
    const tweetInfo = { tweetOwner:"Elon Musk",tweetContent:"Bitcoin is cool"}
    const tree = renderer.create(<CardTweets
        tweetOwner={tweetInfo.tweetOwner}
        tweetContent={tweetInfo.tweetContent}/>).toJSON();
    expect(tree).toMatchSnapshot();
})