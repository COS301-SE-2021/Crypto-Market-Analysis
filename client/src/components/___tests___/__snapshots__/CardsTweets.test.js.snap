// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <link
    href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css"
    rel="stylesheet"
  />,
  <div
    className=" flex flex-col min-w-0 break-words bg-white rounded mb-6 xl:mb-0 shadow-lg"
    data-testid="tweet-card"
  >
    <div
      className="flex-auto p-3"
    >
      <div
        className="flex flex-wrap"
      >
        <div
          className="relative w-full pr-4 max-w-full flex-grow flex-1"
        >
          <h5
            className="text-blueGray-700 uppercase font-bold text-l"
          >
            Elon Musk
          </h5>
        </div>
        <div
          className="relative w-auto pl-4 flex-initial"
        >
          <h3
            className="mb0"
          >
            <i
              className="fa fa-twitter text-info pull-right"
            />
          </h3>
        </div>
      </div>
      <div
        className="relative w-full pr-4 max-w-full flex-grow flex-1"
      >
        <h5
          className="text-blueGray-700  font-bold text-m"
        >
          Tweet
        </h5>
      </div>
      <ul
        className="list-group"
      >
        <li
          className="list-group-item"
        >
          <p
            className="nm"
          >
            Bitcoin is cool
          </p>
          <small
            className="text-muted"
          >
            1 day ago
          </small>
        </li>
      </ul>
    </div>
  </div>,
]
`;
